app/models/firmware_registry.rb

```diff
@@ -32,7 +32,10 @@ def sync_fw_binaries_raw
   end
 
   def self.create_firmware_registry(options)
-    klass = options.delete(:type).constantize
+    type = options.delete(:type)
+    raise ArgumentError, _("Invalid firmware registry type [#{type}]") unless descendants.map(&:name).include?(type)
+
+    klass = type.safe_constantize
```

Comment:
Not sure if this is necessary, since you validated it's a real type already - regular constantize is fine